File: test/Authzi.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common/Initialization/SiloHostBuilder.fs
namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra

open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Hosting               
open Orleans.Configuration;
open Orleans.Hosting;
open System
open System.Net

module SiloHostBuilder =
    let Build (configureDelegate: Action<IServiceCollection>) =
        let builder = 
            HostBuilder()
                .UseEnvironment(Environments.Development)
                .UseOrleans(fun context siloBuilder ->          
                   siloBuilder         
                       .UseLocalhostClustering()         
                       .Configure<ClusterOptions>(fun (options:ClusterOptions) ->         
                           options.ClusterId <- "Orleans.Security.Test"         
                           options.ServiceId <- "ServiceId"         
                           ignore())         
                       .Configure<EndpointOptions>(fun (options:EndpointOptions) ->          
                           options.AdvertisedIPAddress <- IPAddress.Loopback         
                           ignore())
                       .AddMemoryGrainStorage("MemoryGrainStorage")
                       .ConfigureServices(configureDelegate) |> ignore)
        builder.Build()

File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common.CSharp/Initialization/SiloHostBuilder.cs
using System;
using System.Net;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Orleans.Configuration;
using Orleans.Hosting;

namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra;

public static class SiloHostBuilder
{
  public static IHost Build(Action<IServiceCollection> configureDelegate)
  {
    var builder = new HostBuilder()
      .UseEnvironment(Environments.Development)
      .UseOrleans((_, siloBuilder) =>
      {
        siloBuilder
          .UseLocalhostClustering()
          .Configure<ClusterOptions>(options =>
          {
            options.ClusterId = "Orleans.Security.Test";
            options.ServiceId = "ServiceId";
          })
          .Configure<EndpointOptions>(options => options.AdvertisedIPAddress = IPAddress.Loopback)
          .AddMemoryGrainStorage("MemoryGrainStorage")
          .ConfigureServices(configureDelegate);
      });

    return builder.Build();
  }
}
